addons/account/wizard/base_document_layout.py

```diff
@@ -1,13 +1,70 @@
-from odoo import models
+from odoo import api, fields, models
 
 
 class BaseDocumentLayout(models.TransientModel):
     _inherit = 'base.document.layout'
 
+    from_invoice = fields.Boolean(compute='_compute_from_invoice', store=False,)
+    qr_code = fields.Boolean(related='company_id.qr_code', readonly=False)
+    vat = fields.Char(related='company_id.vat', readonly=False,)
+    acc_number = fields.Char(compute='_compute_acc_number', inverse='_inverse_acc_number',)
+
     def document_layout_save(self):
         """Save layout and onboarding step progress, return super() result"""
         res = super(BaseDocumentLayout, self).document_layout_save()
         if step := self.env.ref('account.onboarding_onboarding_step_base_document_layout', raise_if_not_found=False):
             for company_id in self.company_id:
                 step.with_company(company_id).action_set_just_done()
+            if res.get('context') is not None:
+                res['context']['dialog_size'] = 'large'
         return res
+
+    @api.depends('report_layout_id')
+    def _compute_from_invoice(self):
+        move = self.env['account.move'].browse(self.env.context.get('active_id'))
+        for record in self:
+            if self.env.context.get('active_model') == 'account.move':
+                record.from_invoice = move.move_type == 'out_invoice'
+            else:
+                record.from_invoice = False
+
+    @api.depends('qr_code')
+    def _compute_preview(self):
+        """
+            This override is needed to add the depends on the qr code and the move
+        """
+        ir_ui_view, wizard_with_logo, preview_css, is_html_empty = super()._compute_preview()
+        if self.env.context.get('active_model') == 'account.move':
+            move = self.env['account.move'].browse(self.env.context.get('active_id'))
+            # If there is a move in the context then, the value put in the preview come from the move
+            if move:
```

Comment:
This `if` statement is almost useless as we'll use the [dunder method](https://github.com/odoo/odoo/blob/fb0569eeba0a8909a067f2ad90b80edbe8d2d915/odoo/models.py#L6478) `__bool__` from BaseModel, which will be `True` every time as we just create a recordset with the `active_id`, it will be `False` only if `active_id` is not set. 
We can use the `exists` [method](https://github.com/odoo/odoo/blob/fb0569eeba0a8909a067f2ad90b80edbe8d2d915/odoo/models.py#L5654) that will check that the record exists in the database.